### PR TITLE
Change the e2e tests to be sync-to-blockchain

### PR DESCRIPTION
**Pull request by mateo-mro**
# Description
Change the e2e tests to be sync-to-blockchain, so we know the tests are testing the actual configuration.

# Changes
- Modify the autopilot to be sync-to-blockchain
- Modify failing tests (if any)

## How to test
1. Regression tests

### CODE REVIEW

**Comment by sunce86** [comment]
<a href="https://imgflip.com/i/95zu18"></a><div>

File: crates/e2e/tests/e2e/buffers.rs
Comment (MartinquaXD) [reply crates/e2e/tests/e2e/buffers.rs:81]: Why do we need this particular block here anyway?
AFAICS the block mined after creating the new order should be enough.

Comment (MartinquaXD) [reply crates/e2e/tests/e2e/buffers.rs:81]: Also every transaction you execute to set up the state of the chain also mines transactions.

File: crates/e2e/src/setup/services.rs
Comment (MartinquaXD) [inline crates/e2e/src/setup/services.rs:148]: `400ms` seems a bit long. Did smaller values cause issues here?

**Review by MartinquaXD — COMMENTED** [review]
Can you use the [flaky test runner](https://github.com/cowprotocol/services/blob/main/.github/workflows/pull-request.yaml#L184-L219) to see if these tests are stable on CI?

File: crates/e2e/tests/e2e/buffers.rs
Comment (MartinquaXD) [reply crates/e2e/tests/e2e/buffers.rs]: I got your point. I'm just saying it's not clear to me why we need to have this new block right after starting the services if we still need to submit a new order and have that included in an auction before the test can progress.
The auction built from the block right after starting the services should not have all the state we need AFAICS.

File: crates/e2e/src/setup/services.rs
Comment (MartinquaXD) [reply crates/e2e/src/setup/services.rs:148]: I think this is what happens:
1. begin initialization of autopilot
2. autopilot builds `CurrentBlockStream` which sets current block as latest block
3. autopilot finishes initialization
4. if `3` takes longer than `--max-run-loop-delay` autopilot considers the latest block too old and waits for the new one

Because the test code is blocked until the autopilot has reported it's healthy (which requires new blocks to be mined) we are stuck in a deadlock situation.
I think there are a few way to handle this:
1. continuously mine blocks in a background task while the autopilot is starting up until you see an auction being built
2. adjust run loop to skip waiting for a block if you pass for example `--max-run-loop-delay 0s`
3. detect that orderbook and autopilot are up without looking up the latest auction in the DB

Given that all tests so far assumed that the autopilot builds an auction with whatever onchain state was available so far `1` seems the most reasonable choice to me as it would change the least invariants.
That way `--max-run-loop-delay` could become very small and we don't a continuous block mining in the background which gives us better control over when we build auctions.

Comment (MartinquaXD) [reply crates/e2e/src/setup/services.rs:148]: By default anvil only mines blocks on demand (i.e. when submitting a transaction). It can be configured to do auto mining but I think having the tighter control of "manually" deciding when to mine a block can actually be nice for our tests.

File: crates/e2e/src/setup/services.rs
Comment (MartinquaXD) [reply crates/e2e/src/setup/services.rs:331]: That's a perfectly fine solution. Just the code seems a bit clunky. Unless it causes issues I would just always mint a block on every iteration to keep it cleaner.

File: .github/workflows/pull-request.yaml
Comment (MartinquaXD) [inline .github/workflows/pull-request.yaml:188]: Remember to revert this change before merging.

File: crates/e2e/tests/e2e/limit_orders.rs
Comment (MartinquaXD) [inline crates/e2e/tests/e2e/limit_orders.rs:795]: Nit: comment should be moved above the line actually mining the block.

File: crates/e2e/tests/e2e/partially_fillable_pool.rs
Comment (MartinquaXD) [inline crates/e2e/tests/e2e/partially_fillable_pool.rs:126]: Was it required to move the block mining into this awkward scope here?
Would be great if that `|| {mine(); false}` construct could be replaced by simply always mining a block on each iteration (comment applies to multiple places).

**Review by MartinquaXD — APPROVED** [review]
Looks alright to me. Would just like to avoid the awkward scopes if possible.

**Comment by MartinquaXD** [comment]
Could you as a follow up please remove the `--run-loop-mode` and the logic related to the old mode?

File: crates/e2e/src/setup/services.rs
Comment (squadgazzz) [inline crates/e2e/src/setup/services.rs:331]: What do you think about adding a comment explaining that this is required only for the `sync-to-blockchain` runloop mode.

File: crates/e2e/tests/e2e/cow_amm.rs
Comment (squadgazzz) [inline crates/e2e/tests/e2e/cow_amm.rs:342]: I am not sure that `wait_for_condition` makes sense now without mining a new block on each iteration. Should we include the mining inside the loop? That would also avoid the requirement to mine a block manually each time before calling this function.

**Review by squadgazzz — APPROVED** [review]
LGTM

**Pull request merged**
